nbsmoke/lint/__init__.py

```diff
@@ -0,0 +1,101 @@
+"""
+Extend pytest to allow flake checking of notebooks.
+
+A notebooks is first converted to a python script, then checked with
+pyflakes.
+
+IPython magics are supposed to be handled in a useful way, being
+converted to python or skipped as appropriate (depending on their
+interaction with python code). See the magics module for more info.
+"""
+
+import sys
+import warnings
+import io
+import os
+import re
+
+import pytest
+import nbformat
+import nbconvert
+
+from . import _pyflakes
+from . import magics
+
+# list of regexes
+flakes_to_ignore = []
+
+class NBLintError(Exception):
+    pass
+
+class LintNb(pytest.Item):
+
+    def repr_failure(self, excinfo):
+        if excinfo.errisinstance(NBLintError):
+            return excinfo.value.args[0]
+        else:
+            return super(LintNb, self).repr_failure(excinfo)
+
+    def runtest(self):
+        with io.open(self.name,encoding='utf8') as nbfile:
+            nb = nbformat.read(nbfile, as_version=4)
+
+            magics_thing = magics.Thing(
```

Comment:
Yeah, I don't know how to name it because I don't know what it is. The problem is that it's all grown organically as there was no plan about what I wanted out of nbsmoke in the first place, I didn't know pytest, etc, so it was just a case of typing until it worked :)

```
# TODO: suddenly had to make some fns into a class to support blacklists; should rework.
class Thing(object):
```

I've renamed magics thing to magics processor, but that's really just cosmetic.

It can definitely be reorganized. If we're still using it in a year's time, I'll be prepared to fix it ;)